src/signalhandler.cc

```diff
@@ -363,11 +392,15 @@ void InstallFailureSignalHandler() {
   for (size_t i = 0; i < ARRAYSIZE(kFailureSignals); ++i) {
     CHECK_ERR(sigaction(kFailureSignals[i].number, &sig_action, NULL));
   }
+#elif defined(OS_WINDOWS)
+  typedef void (*SignalHandlerPointer)(int); 
+  SignalHandlerPointer previousHandler;  
+  previousHandler = signal(SIGABRT, &FailureSignalHandler); 
```

Comment:
Also note:

I ran into a major problem on Windows with signal handling. Let me reproduce the commit message here, as the commit is going to change when I update the PR:

> On Windows, the signal handler is invoked on the thread on which it was installed, which is not necessarily the thread where the test failed. So instead of using the signal handler to dump the stack trace, which would dump it from the wrong thread, we dump the information immediately.

> Doing this causes us to use a few more frames, so `DumpStackInfo` needed to configurable skip number.

> The nicer approach would be to refactor the `MinimalFormatter` and info dumping functions out of the signal handler and into the logging library, which is a more relevant home for them, but it is less invasive to simply move the two functions Windows needed into the internal namespace.

So the _usual_ code path that a test failure takes now skips signal handler entirely, and goes straight to dumping stack and symbols.